src/main/java/net/dv8tion/jda/core/managers/WebhookManagerUpdatable.java

```diff
@@ -0,0 +1,161 @@
+import net.dv8tion.jda.core.requests.Response;
+import net.dv8tion.jda.core.requests.RestAction;
+import net.dv8tion.jda.core.requests.Route;
+import org.json.JSONObject;
+
+public class WebhookManagerUpdatable
+{
+
+    protected final Webhook webhook;
+
+    protected WebhookField<String> name;
+    protected WebhookField<Icon> avatar;
+    protected WebhookField<TextChannel> channel;
+
+    public WebhookManagerUpdatable(Webhook webhook)
+    {
+        this.webhook = webhook;
+        setupFields();
+    }
+
+    public JDA getJDA()
+    {
+        return webhook.getJDA();
+    }
+
+    public Webhook getWebhook()
+    {
+        return webhook;
+    }
+
+    public WebhookField<String> getNameField()
+    {
+        return name;
+    }
+
+    public WebhookField<Icon> getAvatarField()
+    {
+        return avatar;
+    }
+
+    public WebhookField<TextChannel> getChannelField()
+    {
+        return channel;
+    }
+
+    public void reset()
+    {
+        name.reset();
+        avatar.reset();
+        channel.reset();
+    }
+
+    public RestAction<Void> update()
+    {
+        if (!webhook.getGuild().getSelfMember().hasPermission(webhook.getChannel(), Permission.MANAGE_WEBHOOKS))
+            throw new PermissionException(Permission.MANAGE_WEBHOOKS);
+        if (!shouldUpdate())
+            return new RestAction.EmptyRestAction<>(null);
+
+        JSONObject data = new JSONObject();
+        data.put("name", name.getOriginalValue());
+
+        if (channel.shouldUpdate())
+            data.put("channel_id", channel.getValue().getId());
+        if (name.shouldUpdate())
+            data.put("name", name.getValue());
+        if (avatar.shouldUpdate())
+        {
+            Icon value = avatar.getValue();
+            data.put("avatar", value != null ? value.getEncoding() : JSONObject.NULL);
+        }
+
+        Route.CompiledRoute route = Route.Webhooks.MODIFY_WEBHOOK.compile(webhook.getId());
+        return new RestAction<Void>(getJDA(), route, data)
+        {
+            @Override
+            protected void handleResponse(Response response, Request request)
+            {
+                if (response.isOk())
+                    request.onSuccess(null);
+                else
+                    request.onFailure(response);
+            }
+        };
+    }
+
+    protected boolean shouldUpdate()
+    {
+        return name.shouldUpdate()
+                || avatar.shouldUpdate()
+                || channel.shouldUpdate();
+    }
+
+    protected void setupFields()
+    {
+        name = new WebhookField<String>(this, webhook::getName)
+        {
+            @Override
+            public void checkValue(String value)
+            {
+                checkNull(value, "default name");
+            }
+        };
+
+        avatar = new WebhookField<Icon>(this, null)
+        {
+            @Override
+            public void checkValue(Icon value)
+            {
+                checkNull(value, "default avatar icon");
```

Comment:
Can you verify that you can remove the `avatar` of a Webhook with this code? I know that you said that you based it on pre-existing code, but looking back at that code.. I'm not sure about the validity of it. So do me a favor and test this if you haven't already please.
I'll will be accepting after this has been checked.